packages/fx-core/src/core/middleware/projectUpgrader.ts

```diff
@@ -179,58 +179,18 @@ export async function upgradeContext(ctx: CoreHookContext): Promise<Result<undef
 }
 
 // TODO: add readUserData as basic API in core since used in multiple places.
-async function readUserData(
-  userDataPath: string,
-  projectId?: string
-): Promise<Record<string, string>> {
+async function readUserData(userDataPath: string): Promise<Record<string, string>> {
+  let dict: Record<string, string> = {};
   if (await fs.pathExists(userDataPath)) {
     const dictContent = await fs.readFile(userDataPath, "UTF-8");
-    if (dictContent) {
-      const dict = deserializeDict(dictContent);
-      if (dict && projectId) {
-        const cryptoProvider = new LocalCrypto(projectId);
-        for (const secretKey of Object.keys(dict)) {
-          if (!dataNeedEncryption(secretKey)) {
-            continue;
-          }
-          const secretValue = dict[secretKey];
-          const plaintext = cryptoProvider.decrypt(secretValue);
-          if (plaintext.isErr()) {
-            const fxError: SystemError = plaintext.error;
-            const fileName = basename(userDataPath);
-            fxError.message = `Project update failed because of ${fxError.name}(file:${fileName}):${fxError.message}, if your local file '*.userdata' is not modified, please report to us by click 'Report Issue' button.`;
-            fxError.userData = `file: ${fileName}\n------------FILE START--------\n${dictContent}\n------------FILE END----------`;
-            sendTelemetryErrorEvent(Component.core, TelemetryEvent.DecryptUserdata, fxError);
-            throw plaintext.error;
-          }
-          dict[secretKey] = plaintext.value;
-        }
-        return dict;
-      }
-    }
+    dict = deserializeDict(dictContent);
   }
-  return {};
+
+  return dict;
 }
 
 // TODO: add saveUserData as basic API in core since used in multiple places.
-async function saveUserData(
-  userDataPath: string,
-  userData: Record<string, string>,
-  projectId?: string
-): Promise<void> {
-  if (projectId) {
-    const cryptoProvider = new LocalCrypto(projectId);
-    for (const secretKey of Object.keys(userData)) {
-      if (!dataNeedEncryption(secretKey)) {
-        continue;
-      }
-
-      const encryptedSecret = cryptoProvider.encrypt(userData[secretKey]);
-      if (encryptedSecret.isOk()) {
-        userData[secretKey] = encryptedSecret.value;
-      }
-    }
-  }
+async function saveUserData(userDataPath: string, userData: Record<string, string>): Promise<void> {
```

Comment:
will userData be null or empty?

Reply:
Will not throw error when userdata is empty.

Reply:
serializeDict will handle the cipher work right?

Reply:
Here we will not handle the cipher. What we do is just read and write and we don't care about whether it is encrypted.